From 73e4fbc5777eddd89bb0fb16e90343551fe3e0ef Mon Sep 17 00:00:00 2001
From: Jussi Kukkonen <jussi.kukkonen@intel.com>
Date: Tue, 5 Jul 2016 10:56:23 +0300
Subject: [PATCH] cmake: Link utils with xcb explicitly

Linking with g++ can reportedly lead to:

libpiglitutil_gl.so.0: error: undefined reference to 'xcb_connect'
libpiglitutil_gl.so.0: error: undefined reference to 'xcb_get_setup'
libpiglitutil_gl.so.0: error: undefined reference to 'xcb_setup_roots_iterator'

This may have appeared now because xcb-dri2 used to overlink publicly
but now does not.

Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
Upstream-Status: Accepted
---
 CMakeLists.txt            | 1 +
 tests/util/CMakeLists.txt | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e2abba..2e1a473 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,6 +151,7 @@ IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 
 	pkg_check_modules(LIBDRM QUIET libdrm)
 	pkg_check_modules(LIBDRM_INTEL QUIET libdrm_intel)
+	pkg_check_modules(XCB QUIET xcb)
 	pkg_check_modules(XCB_DRI2 QUIET xcb-dri2)
 	pkg_check_modules(GLPROTO QUIET glproto)
 ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
diff --git a/tests/util/CMakeLists.txt b/tests/util/CMakeLists.txt
index fb22ffa..e1f8073 100644
--- a/tests/util/CMakeLists.txt
+++ b/tests/util/CMakeLists.txt
@@ -121,11 +121,13 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 
 		list(APPEND UTIL_GL_LIBS
 			${LIBDRM_LDFLAGS}
+			${XCB_LDFLAGS}
 			${XCB_DRI2_LDFLAGS}
 		)
 
 		list(APPEND UTIL_GL_INCLUDES
 			${LIBDRM_INCLUDE_DIRS}
+			${XCB_INCLUDE_DIRS}
 			${XCB_DRI2_INCLUDE_DIRS}
 		)
 
-- 
2.1.4

